### PR TITLE
feat: add `bv_decide` normalization rules for `ofBool (a.getLsbD i)` and `ofBool a[i]`

### PR DESCRIPTION
**Pull request by alexkeizer**
In LNSym we often use the pattern `ofBool (a.getLsbD i)` to pick out a specific bit (`i`) from a bitvector (`a`).

By adding a rewrite to `extractLsb` to `bv_decide`s normalization set, we can still automatically close goals that have this pattern. In the process, I also added a simp-lemma about the value of a `Fin 1`.

### CODE REVIEW

File: src/Std/Tactic/BVDecide/Normalize/BitVec.lean [hunk 1]
```diff
@@ -206,5 +206,17 @@ theorem BitVec.max_ult' (a : BitVec w) : (BitVec.ult (-1#w) a) = false := by
...
+@[bv_normalize]
+theorem BitVec.ofBool_getLsbD (a : BitVec w) (i : Nat) :
+    BitVec.ofBool (a.getLsbD i) = a.extractLsb' i 1 := by
+  ext j
+  simp
+
+@[bv_normalize]
+theorem BitVec.ofBool_getElem (a : BitVec w) (i : Nat) (h : i < w) :
+    BitVec.ofBool a[i] = a.extractLsb' i 1 := by
+  rw [show a[i] = a.getLsbD i from rfl]
```
Comment (hargoniX) [inline src/Std/Tactic/BVDecide/Normalize/BitVec.lean:218]: There should probably exist a lemma for this? If it doesn't I think we should add it?

File: tests/lean/run/bv_extract.lean [hunk 2]
```diff
@@ -13,3 +13,15 @@ theorem bv_extract_3 (h : x = 1#64) : extractLsb 63 32 x = 0#32 := by
...
 theorem bv_extract_4 (h : x = 1#64) : extractLsb 31 0 x = 1#32 := by
   bv_decide
+
+theorem bv_ofBool_1 (h : x = 1#64) : ofBool (x.getLsbD 0) = 1#1 := by
+  bv_decide
+
+theorem bv_ofBool_2 (h : x = 1#64) : ofBool (x.getLsbD 1) = 0#1 := by
+  bv_decide
+
+theorem bv_ofBool_3 (h : x = 1#64) : ofBool (x[0]) = 1#1 := by
```
Comment (hargoniX) [inline tests/lean/run/bv_extract.lean:23]: No need for parenthesis here I think?

Comment (alexkeizer) [reply tests/lean/run/bv_extract.lean:23]: Indeed not

File: src/Std/Tactic/BVDecide/Normalize/BitVec.lean (hunk 1 above)
Comment (alexkeizer) [reply src/Std/Tactic/BVDecide/Normalize/BitVec.lean]: You're right, I didn't find it cause my master was outdated.

I did golf the proof of the lemma while I was at it: it used to involve `simp`, but `rfl` is enough.

**Comment by alexkeizer** [comment]
@hargoniX addressed your comments, another round of review please!

**Review by tobiasgrosser — APPROVED** [review]
Nice. Thank you.

**Review by hargoniX — APPROVED** [review]

**Pull request merged**